circle/attestation.go

```diff
@@ -1,27 +1,39 @@
 package circle
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 	"time"
 
 	"cosmossdk.io/log"
+
 	"github.com/strangelove-ventures/noble-cctp-relayer/types"
 )
 
 // CheckAttestation checks the iris api for attestation status and returns true if attestation is complete
-func CheckAttestation(attestationURL string, logger log.Logger, irisLookupId string, txHash string, sourceDomain, destDomain types.Domain) *types.AttestationResponse {
-	logger.Debug(fmt.Sprintf("Checking attestation for %s%s%s for source tx %s from %d to %d", attestationURL, "0x", irisLookupId, txHash, sourceDomain, destDomain))
+func CheckAttestation(attestationURL string, logger log.Logger, irisLookupID string, txHash string, sourceDomain, destDomain types.Domain) *types.AttestationResponse {
+	logger.Debug(fmt.Sprintf("Checking attestation for %s%s%s for source tx %s from %d to %d", attestationURL, "0x", irisLookupID, txHash, sourceDomain, destDomain))
 
 	client := http.Client{Timeout: 2 * time.Second}
 
-	rawResponse, err := client.Get(attestationURL + "0x" + irisLookupId)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, attestationURL+"0x"+irisLookupID, nil)
```

Comment:
Only worry here is that attestationURL may or may not have a / on the end (easy to forget in future code). Maybe add a check here that if attestationURL does not HasSuffix "/", append that to attestationURL

same for irisLookupID? (if !HasPrefix 0x, pre-pend

Reply:
Good catch! Just added some checks to `attestation.go` and created a couple unit tests to validate it's working.